### PR TITLE
Go out

### PR DESCRIPTION
**Pull request by meowjesty**
Support outgoing traffic for golang.

### CODE REVIEW

File: tests/src/sanity.rs [hunk 1]
```diff
@@ -965,4 +965,16 @@ mod tests {
         assert!(res.success());
         process.assert_stderr();
     }
+
+    #[rstest]
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    pub async fn test_go_outgoing_traffic_single_request_enabled(#[future] service: EchoService) {
+        let service = service.await;
+        let command = vec!["go-e2e-outgoing/go-e2e-outgoing"];
```
Comment (aviramha) [inline tests/src/sanity.rs:973]: add 18/19 versions (see PR that's about to be merged #365

File: mirrord-layer/src/socket/ops.rs [hunk 2]
```diff
@@ -216,6 +218,59 @@ pub(super) fn connect(sockfd: RawFd, remote_address: SocketAddr) -> Result<()> {
         .get()
         .expect("Should be set during initialization!");
 
+    if let SocketKind::Udp(kind) = user_socket_info.kind {
+        // Prepare this socket to be intercepted.
```
Comment (aviramha) [inline mirrord-layer/src/socket/ops.rs:222]: bypass if not port `53`

**Review by aviramha — COMMENTED** [review]
Looking good! Just 2 comments. see inline.

File: tests/src/sanity.rs (hunk 1 above)
Comment (meowjesty) [reply tests/src/sanity.rs]: Added.

File: mirrord-agent/src/outgoing/udp.rs [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+        .parse()
+        .map_err(RemoteError::from)?;
+
+    Ok(dns_address)
+}
+
+async fn connect(remote_address: SocketAddr) -> Result<UdpSocket, ResponseError> {
+    trace!("connect -> remote_address {:#?}", remote_address);
+
+    let remote_address = if remote_address.port() == 53 {
```
Comment (aviramha) [inline mirrord-agent/src/outgoing/udp.rs:83]: use const you added forthat?

**Review by aviramha — CHANGES_REQUESTED** [review]
nitpick + UDP is always enabled currently - add a flag to enable/disable it (we'll have it enabled by default..)

**Review by aviramha — APPROVED** [review]
LGTM! Let's get this merged 🥳

**Pull request merged**